### ISSUE TITLE
Add automatic BNS name registration handling

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#2754]
To smoothen out the process for registering BNS names

For example:

- Automatically queue up `name-register` calls for the user upon broadcasting of `name-preorder`
- Automatically retry `name-register` calls if they first fail

Requested during a Twitter Space with the community in October 2022 

cc @bilalanees98 for input on whether these (or other) wallet enhancements would help with common registration problems

**Comment by kyranjamie** [comment]
We should extremely careful about—or completely avoid—implementing features like this:

- Why should BNS get special treatment compared with other contracts?
- What if a user doesn't want it to retry? We should never assume such.
- It isn't possible with Ledger
- It won't work if the users closes their browser/their computer sleeps

It should be the responsibility of an app to take care of this. They won't be able to automate it completely, but the experience could still be improved I'm sure.

**Comment by bilalanees98** [comment]
I agree with all the points that @kyranjamie has raised. Additionally there will be technical challenges involved.

To answer the initial questions:

> Automatically queue up name-register calls for the user upon broadcasting of name-preorder

For the `name-preorder` a `salt` is necessary. Its used to create the salted hash of the name that is broadcast. (This is what stops other users from front-running the domain in the preorder step). This salt needs to be broadcast during the register step. For btc.us this is randomly calculated and stored on gaia. Since, it is not possible for another application to access btc.us specific data on gaia then this info will have to be passed to the wallet and I am not sure if this is possible.

> Automatically retry name-register calls if they first fail

Automatic retrying won't fix the reason why the tx failed. e.g. if the register tx failed because the preorder was not found then retrying will not do anything. Automating any process to cater to network uncertainty entirely is not possible imo.

**Comment by aulneau** [comment]
I don't think the wallet should ever automatically re-submit transactions, but I do think the wallet should support BNS functionality built in. If we're thinking about it from the users perspective, BNS is a built-in primitive that has special treatment in the ecosystem/network, and wallets should provide special treatment to those transactions as they help push the overall experience forward in many positive ways.
